pkg/registry/servicecatalog/clusterserviceclass/strategy.go

```diff
@@ -0,0 +1,153 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterserviceclass
+
+// this was copied from where else and edited to fit our objects
+
+import (
+	"github.com/kubernetes-incubator/service-catalog/pkg/api"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/apiserver/pkg/storage/names"
+
+	"github.com/golang/glog"
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	scv "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/validation"
+)
+
+// NewScopeStrategy returns a new NamespaceScopedStrategy for cluster service
+// classes.
+func NewScopeStrategy() rest.NamespaceScopedStrategy {
+	return clusterServiceClassRESTStrategies
+}
+
+// clusterServiceClassRESTStrategy implements interfaces RESTCreateStrategy,
+// RESTUpdateStrategy, RESTDeleteStrategy, NamespaceScopedStrategy.
+type clusterServiceClassRESTStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+	names.NameGenerator // GenerateName method for CreateStrategy
+}
+
+// clusterServiceClassStatusRESTStrategy implements interface
+// RESTUpdateStrategy. This implementation validates updates to
+// clusterServiceClass.Status updates only and disallows any modifications to
+// the clusterServiceClass.Spec.
+type clusterServiceClassStatusRESTStrategy struct {
+	clusterServiceClassRESTStrategy
+}
+
+var (
+	clusterServiceClassRESTStrategies = clusterServiceClassRESTStrategy{
+		// embeds to pull in existing code behavior from upstream
+
+		ObjectTyper: api.Scheme,
+		// use the generator from upstream k8s, or implement method
+		// `GenerateName(base string) string`
+		NameGenerator: names.SimpleNameGenerator,
+	}
+	_ rest.RESTCreateStrategy = clusterServiceClassRESTStrategies
+	_ rest.RESTUpdateStrategy = clusterServiceClassRESTStrategies
+	_ rest.RESTDeleteStrategy = clusterServiceClassRESTStrategies
+
+	clusterServiceClassStatusUpdateStrategy = clusterServiceClassStatusRESTStrategy{
+		clusterServiceClassRESTStrategies,
+	}
+	_ rest.RESTUpdateStrategy = clusterServiceClassStatusUpdateStrategy
+)
+
+// Canonicalize does not transform a ClusterServiceClass.
+func (clusterServiceClassRESTStrategy) Canonicalize(obj runtime.Object) {
+	_, ok := obj.(*sc.ClusterServiceClass)
+	if !ok {
+		glog.Fatal("received a non-clusterserviceclass object to create")
+	}
+}
+
+// NamespaceScoped returns false as ClusterServiceClass are not scoped to a
+// namespace.
+func (clusterServiceClassRESTStrategy) NamespaceScoped() bool {
```

Comment:
I don't know what the rest of k8s does, but maybe this is called just `Namespaced` ?

Reply:
🤷‍♂️